pkg/controller/core/cohort_controller.go

```diff
@@ -82,6 +126,16 @@ func (r *CohortReconciler) Update(e event.UpdateEvent) bool {
 }
 
 func (r *CohortReconciler) Delete(e event.DeleteEvent) bool {
+	cohort, ok := e.Object.(*kueue.Cohort)
+	if !ok {
+		return false
+	}
+
+	log := r.log.WithValues("cohort", klog.KObj(cohort))
```

Comment:
what is the motivation for the move? My understanding of Reconciler is these functions (Create/Update/Delete) is just a filter, and we shouldn't be executing logic in it (I know we do this for other reconcilers, maybe can fix at some point)

Reply:
> what is the motivation for the move?

When executing CohortAncestors() we can return a cohort without an object, and in this case, we’ll remove it from the cache since it’s unnecessary.

Reply:
A race condition? Can we just ignore this Cohort, since another reconcile should cleanup that Cohort anyway from Cache (right?)?

Reply:
Not exactly. We have a `kueuealpha.Cohort` called "cohort" with a parent "root", which is just a string and doesn’t have any resources. CohortAncestors() returns `[]string{"root"}`, but now the Reconciler deletes it from the cache, which isn’t entirely correct.

Reply:
Ahh, this is the implicit cohort mechanism? In this case, what if we don't return the root from ancestors (and document this in the function comment)

As the root will not have a FairSharing value anyway: https://github.com/kubernetes-sigs/kueue/blob/a454f0e133682887c8043b10cc360bc24f7d3924/pkg/cache/fair_sharing.go#L48-L50

Reply:
> Ahh, this is the implicit cohort mechanism? In this case, what if we don't return the root from ancestors (and document this in the function comment)

But what if root is really `kueuealpha.Cohort` object?

Reply:
In both cases, the FairSharing score isn't relevant for this Cohort = it is always 0 - so I think we can skip triggering the reconcile

At some point this Cohort will be reconciled, and it will be given a score of 0 - so that should be consistent from the UI perspective

Reply:
Done